tests/lib/common.py

```diff
@@ -56,11 +56,14 @@ def wait_for_result(func, *args, matcher=simple_matcher(True), attempts=20,
     """
 
     for i in range(attempts):
-        out = func(*args, **kwargs)
-        if decode:
-            out = decode(out)
-        if matcher(out):
-            return out
+        try:
+            out = func(*args, **kwargs)
+            if decode:
+                out = decode(out)
+            if matcher(out):
+                return out
+        except Exception:
+            pass
```

Comment:
We could just log the last exception rather than every one since that might be too noisy is the func is raising something at each attempt.

Lets also add an option to `wait_for_result` to choose whether to catch exceptions. (eg `catch_exceptions=True`)

Reply:
For example below, the function call returns an error, and wait_for_result() exists without retry.
The previous code only handled result of function() but not a failure during execution of function()

```
11:04:16                           tests.lib.rook.base    INFO : Wait for Ceph HEALTH_OK
11:04:16 kubectl -n rook-ceph get pod -l app="rook-cep    INFO : rook-ceph-tools-7f849c7dd9-jvjwj
11:04:16 kubectl -n rook-ceph exec -t "rook-ceph-tools WARNING : unable to get monitor info from DNS SRV with service name: ceph-mon
11:04:16 kubectl -n rook-ceph exec -t "rook-ceph-tools WARNING : [errno 2] RADOS object not found (error connecting to the cluster)
11:04:16 kubectl -n rook-ceph exec -t "rook-ceph-tools WARNING : command terminated with exit code 1
11:04:16                           tests.lib.rook.base    INFO : rook destroy on <tests.lib.hardware.libvirt.Hardware object at 0x7fbf24edb6a0>
```

Reply:
Added option as requested